src/imperative/imperative_utils.cc

```diff
@@ -120,6 +121,18 @@ void InvokeOperator(const nnvm::IndexedGraph& idx,
       *arrays[eid] = NDArray();
     }
   }
+  if (callback) {
+    for (size_t i = 0; i < ndinputs.size(); ++i) {
+      NDArray* cpy = new NDArray(*ndinputs[i]);
+      callback(node.source->attrs.name.c_str(), node.source->op()->name.c_str(),
+               reinterpret_cast<void*>(cpy));
+    }
+    for (size_t i = 0; i < ndoutputs.size(); ++i) {
+      NDArray* cpy = new NDArray(*ndoutputs[i]);
+      callback(node.source->attrs.name.c_str(), node.source->op()->name.c_str(),
```

Comment:
i guess first is tensor name, 2nd is operator name, 3rd is value. Can you confirm this?
Is there a way to distinguish if the tensor is input or output ?

Also, i don't see monitor_all being used, is it used somewhere implicitly not in this PR?

Reply:
yes you are right. there is a way to distinguish by having a postfix after node name for outputs. Alternatively we can have different callback registration apis for inputs and outputs. Currently it is common for both.

monitor_all is not being used yet and the intention was to use it for inputs vs inputs and outputs, similar to the way it is done with graph executor.